### DISCUSSION TITLE
[Q&A] Can't discover FS Switch with SNMP

### DISCUSSION THREAD

**Discussion by KingObal** [glpi-project/glpi-agent#606] [Q&A] [answered]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

Hello everyone,

I'm using GLPI to manage several entities and recently encountered a problem with switch discovery. I've already managed to bring up switches with SNMP using tasks in several entities, but I'm now stuck for one specific entity.

Out of a dozen FS switches, only the 2 network cores and 2 NASs come up and can be inventoried. I've checked that SNMP is enabled on all switches with the right community, and I've also enabled LLDP. Despite this, I can't find them.

I've tried running the NetDiscovery and NetInventory programs in debug mode from the server with the discovery modules, and they work fine. All the correct VLANs come up in the inventory.

However, when I run the discovery tasks, only the 2 cores and the 2 NAS are discovered, Switches get Added as unmanaged equipement.

GLPI ver : 10.0.12
GLPI-Agent ver : 1.7.1

Here is the debug output of the net-discovery program for one switch IP address :

```
[debug] #1, scanning 10.10.XX.XX
[debug] #1, partial match for sysobjectID .1.3.6.1.4.1.52642.1.99.4872 on manufacturer ID
[debug] #1, - scanning 10.10.XX.XX with SNMP, credentials 1: success
[debug] #1, - scanning 10.10.XX.XX with netbios: no result
[debug] #1, - scanning 10.10.XX.XX with echo ping: success
[debug] #1, - scanning 10.10.XX.XX in arp table: no result
<?xml version="1.0" encoding="UTF-8"?>
<REQUEST>
  <CONTENT>
    <DEVICE>
      <AUTHSNMP>1</AUTHSNMP>
      <CONTACT>opentools system</CONTACT>
      <DESCRIPTION>FSOS software, S5850 software (S5850 24S2Q), Version 7.4.3.r2
Copyright (C) 2009-2022 FS.COM Inc.  All Rights Reserved.
Compiled Jun 30 08:34:17 2022</DESCRIPTION>
      <DNSHOSTNAME>10.10.XX.XX</DNSHOSTNAME>
      <FIRMWARE>7.4.3.r2</FIRMWARE>
      <IP>10.10.XX.XX</IP>
      <IPS>
        <IP>10.10.XX.XX</IP>
      </IPS>
      <LOCATION>A0</LOCATION>
      <MAC>64:9d:XX:XX:XX:XX</MAC>
      <MANUFACTURER>FS</MANUFACTURER>
      <MODEL>24S2Q</MODEL>
      <SERIAL>CG220418852XXXXX</SERIAL>
      <SNMPHOSTNAME>A0-C1</SNMPHOSTNAME>
      <TYPE>NETWORKING</TYPE>
      <UPTIME>124 days, 14:11:31.00</UPTIME>
    </DEVICE>
    <MODULEVERSION>6.1</MODULEVERSION>
    <PROCESSNUMBER>1</PROCESSNUMBER>
  </CONTENT>
  <DEVICEID>foo</DEVICEID>
  <QUERY>NETDISCOVERY</QUERY>
</REQUEST>
[debug] #1, #1, worker termination 
```

Thanks a lot for your help !!!

**Comment by KingObal** [comment]
I managed to solve my problem by adding the OID to the SysObject.ids file, there was only one FS OID.

**Reply by g-bougard** [reply, marked as answer]
Hi @KingObal 

as your outpuot is showing, there was still a `partial match for sysobjectID`, and this should have been sufficient. I don't see what could have change in the code logic after you have added your entry.

What line did you exactly added to the file ?

**Reply by KingObal** [reply]
i'm sorry this was the wrong switch output. 

It was a partial match on all switches but only this one was working.
On the others FS swich i had no serial, so it was not importing the switches. 

i wanted to close this request, 
i opened this one https://github.com/glpi-project/sysobject.ids/issues/27

i'm sorry to bother you on this post !
